### PR TITLE
Unit-Test Workflows and CodeQL workflows

### LINKED ISSUE #10: Setup GitHub Actions for CI on PR and Push Events
There is no continuous integration (CI) setup for testing the code when a PR is opened or new code is pushed.

Setting up GitHub Actions for CI will help maintain code quality and catch issues early.

This is dependent on the Unit testing issue - #5, which can be utilized for testing in the CI.

### PR DESCRIPTION
**Pull request by vihar-s1**
# Pull Request Overview

### unit-tests.yml workflow

- Github workflow file for running unit-tests when a new PR is raised, or code is pushed to the main branch

### codeql.yml workflow

- Workflow file for performing static analysis on the codebase for a new PR or a code push to main branch

### CODE REVIEW

**Comment by vihar-s1** [comment]
Sorry for the delay...was pr-occupied with some other task...please tell me if you require some additional changes

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            - name: Autobuild
+              uses: github/codeql-action/autobuild@v2
+            # - run: |
+            #     echo "Run, Build Application using script"
+            #     ./location_of_script_within_repo/buildscript.sh
+
+            - name: Perform CodeQL Analysis
+              uses: github/codeql-action/analyze@v2
+              with:
+                  category: "/language:${{matrix.language}}"
```
Comment (Delta456) [inline .github/workflows/codeql.yml:45]: Add newline here

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            - name: Checkout Repo
+              uses: actions/checkout@v4
+            - name: Set up JDK 17
+              uses: actions/setup-java@v4
+              with:
+                  distribution: 'oracle'
+                  java-version: '17'
+            - name: Setup Gradle
+              uses: gradle/actions/setup-gradle@v3
+            - name: Run test suites
```
Comment (Delta456) [inline .github/workflows/unit-tests.yml:26]: Add newline here as well

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (vihar-s1) [reply .github/workflows/codeql.yml:45]: sure

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (vihar-s1) [reply .github/workflows/unit-tests.yml:26]: you want a new line at the end right ??

Comment (Delta456) [reply .github/workflows/unit-tests.yml:26]: Yes

Comment (vihar-s1) [reply .github/workflows/unit-tests.yml]: added blank lines at end for both the files

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ "java-kotlin" ]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```
Comment (harsha509) [inline .github/workflows/codeql.yml:26]: can we use `actions/checkout@v4`

File: .github/workflows/codeql.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+          queries: security-extended,security-and-quality
+
+      # Autobuild attempts to build any compiled languages (C/C++, C#, Go, Java, or Swift).
+      # If this step fails, then you should remove it and run the build manually (see below)
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@v2
+      # - run: |
```
Comment (harsha509) [inline .github/workflows/codeql.yml:38]: Please remove dead code!

File: .github/workflows/codeql.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ "java-kotlin" ]
```
Comment (harsha509) [inline .github/workflows/codeql.yml:22]: trying to understabd in here. can we have  ??
```suggestion
        language: [ "java", "kotlin" ]
```

File: .github/workflows/unit-tests.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: "oracle"
+          java-version: "17"
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+
+      - name: Run test suites
```
Comment (harsha509) [inline .github/workflows/unit-tests.yml:29]: Do we require building project before running the tests ??
`./gradlew build`

File: .github/workflows/unit-tests.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        run: ./gradlew test
```
Comment (harsha509) [inline .github/workflows/unit-tests.yml:30]:
```suggestion
        run: ./gradlew test --info
```

For logs!

File: .github/workflows/unit-tests.yml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
```
Comment (harsha509) [inline .github/workflows/unit-tests.yml:12]: can you add name for unit-tests like `name:Run Unit Tests` etc

**Review by harsha509 — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql.yml (hunk 3 above)
Comment (vihar-s1) [reply .github/workflows/codeql.yml:26]: I saw the action results...will make the fixes/changes to resolve the warnings found in thr run

File: .github/workflows/codeql.yml (hunk 5 above)
Comment (vihar-s1) [reply .github/workflows/codeql.yml:22]: i think so...couldn't find anything exactly matching but it should work.

File: .github/workflows/unit-tests.yml (hunk 7 above)
Comment (vihar-s1) [reply .github/workflows/unit-tests.yml:30]: sure

File: .github/workflows/unit-tests.yml (hunk 8 above)
Comment (vihar-s1) [reply .github/workflows/unit-tests.yml:12]: will do!

**Comment by vihar-s1** [comment]
If there are no open issues with the PR, then can you guys get this PR merged? Also, can you please add the `hacktoberfest` label to this PR?

**Review by Delta456 — APPROVED** [review]

**Review by harsha509 — APPROVED** [review]
LGTM!

**Pull request merged**